File: Srims.Server.Business/Common/AnnouncementState.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Srims.Server.Business.Common
{
    /// <summary>
    /// 通知状态
    /// </summary>
    public enum AnnouncementState
    {
        /// <summary>
        /// 未知
        /// </summary>
        Unknown = 0,
        /// <summary>
        /// 正常
        /// </summary>
        Normal = 1,
        /// <summary>
        /// 置顶
        /// </summary>
        Top = 2,
        /// <summary>
        /// 过期
        /// </summary>
        Overdue = 3,
    }
}

File: Srims.WebSite/javascript/srims/common/AnnouncementState.js

if (!Srims.common) 
    Ext.namespace('Srims.common');

Srims.common.AnnouncementState = function(){
};

Srims.common.AnnouncementState.unknown = 'Unknown';
Srims.common.AnnouncementState.Normal = 'Normal';
Srims.common.AnnouncementState.Top = 'Top';
Srims.common.AnnouncementState.Overdue = 'Overdue';

Srims.common.AnnouncementState.render = function(value, metadata){
    switch (value) {
        case 'Unknown':
            return '未知';
        case 'Normal':
            return '正常';
        case 'Top':
            return '置顶';
        case 'Overdue':
            return '过期';
        default:
            return '未知';
    }
}
Srims.common.AnnouncementState.filterItems = [{
    id: 'Unknown',
    text: '未知'
}, {
    id: 'Normal',
    text: '正常'
}, {
    id: 'Top',
    text: '置顶'
}, {
    id: 'Overdue',
    text: '过期'
}];

Srims.common.AnnouncementState.store = [['Unknown', '未知'], ['Normal', '正常'], ['Top', '置顶'], ['Overdue', '过期']];
Srims.common.AnnouncementState.editStore = [['Normal', '正常'], ['Top', '置顶'], ['Overdue', '过期']];
